mantidproject/mantid
# Load isis live timeout for histogram data

This has added a timeout to the ISIS Histogram DAE listener.  Unfortunately the timeout cannot seperate network failure from long running successful operations, so the value of the timeout has been set to 120 seconds by default.

You can configure it in the user.properties file with the key:
```
ISISDAE.Timeout = 20 #seconds
```

**To test:**

**NOTE: This has some windows/linux specific code, it has been developed on windows so I'm pretty confident about it working their, but it needs testing on Linux/Mac.**

1. Run Mantid, Load->Live Data
1. Pick an instrument ("Offspec", "Engin-X" etc avoid the really big ones like MAPS, MERLIN)
1. Use Histogram mode
1. Rest of the load live settings do not matter much.
1. Run the live load.
1. After the dataset has first appeared pull out the network cable from your PC.
1. Mantid should remain responsive, the monitorLiveData algorithm should remain until the timeout, when you will also get an error message (not a great one, but ok) to the log, then it should disappear.
1. Mantid should close cleanly (after the timeout) without hanging.

1. You should add this line to your user.properties file and repeat the test above, this time the timeout should be only 20 seconds. You will need to restart mantid for the change to be picked up.
```
ISISDAE.Timeout = 20 #seconds
```

Fixes #20746
**Release Notes** 
docs/source/release/.../framework.rst

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
